src/backend/dx12/src/format.rs

```diff
@@ -0,0 +1,1123 @@
+//! Format related queries for the backend.
+
+use hal::format::{BufferFeature, ImageFeature, Properties, NUM_FORMATS};
+
+///
+pub fn query_properties() -> [Properties; NUM_FORMATS] {
```

Comment:
This would be somewhat nice to just load from a RON or something, since it's pure data, rather verbose, and structurally simple. It's blocked by the bitflags support though.

Reply:
to some extent yes, but we also need to manually check for optional format properties on physical device enumeration.